android-sdk/android/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt

```diff
@@ -201,18 +222,36 @@ open class LogtoClient(
                     )
                     return@fetchTokenByRefreshToken
                 }
-                requireNotNull(refreshTokenTokenResponse).let { tokenResponse ->
+
+                getJwks { getJwksException, jwks ->
+                    getJwksException?.let {
+                        completion.onComplete(it, null)
+                        return@getJwks
+                    }
+                    val refreshedToken = requireNotNull(fetchedTokenResponse)
+                    refreshedToken.idToken?.let {
+                        try {
+                            TokenUtils.verifyIdToken(it, logtoConfig.clientId, oidcConfig.issuer, requireNotNull(jwks))
+                        } catch (exception: InvalidJwtException) {
+                            completion.onComplete(
+                                LogtoException(LogtoException.Message.INVALID_ID_TOKEN, exception),
+                                null
+                            )
+                            return@getJwks
+                        }
+                    }
+
                     val refreshedAccessToken = AccessToken(
-                        token = tokenResponse.accessToken,
-                        scope = tokenResponse.scope,
+                        token = refreshedToken.accessToken,
+                        scope = refreshedToken.scope,
                         expiresAt = expiresAtFrom(
                             nowRoundToSec(),
-                            tokenResponse.expiresIn
+                            refreshedToken.expiresIn
                         )
                     )
```

Comment:
looks similar logic for both refreshToken response handler and fetchToken response. Would it possible to extract them out?

Reply:
good idea, updated